app/controllers/messages_controller.rb

```diff
@@ -33,7 +33,10 @@ def create
       render action: :preview and return
     end
 
-    unless @message.valid? && flash.now[:error].nil?
+    if @message.valid? && flash.now[:error].nil? && @message.save
```

Comment:
Should be able to condense this into `if flash.now[:error].nil? && @message.save`